src/routes/Main.tsx

```diff
@@ -0,0 +1,559 @@
+      subtitle: 'BBQ',
+      title: 'BBQ 양념치킨+크림치즈볼+콜라1.25L',
+      imageSrc:
+        'https://st.kakaocdn.net/product/gift/product/20231030175450_53e90ee9708f45ffa45b3f7b4bc01c7c.jpg',
+      amount: 29000,
+      category: '남성이',
+      option: '받고 싶어한',
+    },
+    {
+      rankingIndex: 14,
+      subtitle: 'BBQ',
+      title: 'BBQ 양념치킨+크림치즈볼+콜라1.25L',
+      imageSrc:
+        'https://st.kakaocdn.net/product/gift/product/20231030175450_53e90ee9708f45ffa45b3f7b4bc01c7c.jpg',
+      amount: 29000,
+      category: '남성이',
+      option: '받고 싶어한',
+    },
+    {
+      rankingIndex: 15,
+      subtitle: 'BBQ',
+      title: 'BBQ 양념치킨+크림치즈볼+콜라1.25L',
+      imageSrc:
+        'https://st.kakaocdn.net/product/gift/product/20231030175450_53e90ee9708f45ffa45b3f7b4bc01c7c.jpg',
+      amount: 29000,
+      category: '남성이',
+      option: '받고 싶어한',
+    },
+    {
+      rankingIndex: 16,
+      subtitle: 'BBQ',
+      title: 'BBQ 양념치킨+크림치즈볼+콜라1.25L',
+      imageSrc:
+        'https://st.kakaocdn.net/product/gift/product/20231030175450_53e90ee9708f45ffa45b3f7b4bc01c7c.jpg',
+      amount: 29000,
+      category: '남성이',
+      option: '받고 싶어한',
+    },
+    {
+      rankingIndex: 17,
+      subtitle: 'BBQ',
+      title: 'BBQ 양념치킨+크림치즈볼+콜라1.25L',
+      imageSrc:
+        'https://st.kakaocdn.net/product/gift/product/20231030175450_53e90ee9708f45ffa45b3f7b4bc01c7c.jpg',
+      amount: 29000,
+      category: '남성이',
+      option: '받고 싶어한',
+    },
+    {
+      rankingIndex: 18,
+      subtitle: 'BBQ',
+      title: 'BBQ 양념치킨+크림치즈볼+콜라1.25L',
+      imageSrc:
+        'https://st.kakaocdn.net/product/gift/product/20231030175450_53e90ee9708f45ffa45b3f7b4bc01c7c.jpg',
+      amount: 29000,
+      category: '남성이',
+      option: '받고 싶어한',
+    },
+    {
+      rankingIndex: 19,
+      subtitle: 'BBQ',
+      title: 'BBQ 양념치킨+크림치즈볼+콜라1.25L',
+      imageSrc:
+        'https://st.kakaocdn.net/product/gift/product/20231030175450_53e90ee9708f45ffa45b3f7b4bc01c7c.jpg',
+      amount: 29000,
+      category: '남성이',
+      option: '많이 선물한',
+    },
+    {
+      rankingIndex: 20,
+      subtitle: 'BBQ',
+      title: 'BBQ 양념치킨+크림치즈볼+콜라1.25L',
+      imageSrc:
+        'https://st.kakaocdn.net/product/gift/product/20231030175450_53e90ee9708f45ffa45b3f7b4bc01c7c.jpg',
+      amount: 29000,
+      category: '남성이',
+      option: '많이 선물한',
+    },
+    {
+      rankingIndex: 21,
+      subtitle: 'BBQ',
+      title: 'BBQ 양념치킨+크림치즈볼+콜라1.25L',
+      imageSrc:
+        'https://st.kakaocdn.net/product/gift/product/20231030175450_53e90ee9708f45ffa45b3f7b4bc01c7c.jpg',
+      amount: 29000,
+      category: '남성이',
+      option: '많이 선물한',
+    },
+  ];
+
+  const filteredRankingItems = rankingItems
+    .filter((item) => item.category === selectedPersonCategory || selectedPersonCategory === 'All')
+    .filter((item) => item.option === selectedOption);
+
+  const itemsToShow = showAllRankingItems ? filteredRankingItems : filteredRankingItems.slice(0, 6);
```

Comment:
카테고리 필터링을 어떻게 해야할지 고민이라고 하셨는데 이렇게 1, 2뎁스 필터링을 한번에 해도 좋습니다. 다만 filter를 한번만 쓰거나, showAllRankingItems값이 카테고리 전체 체크여부가 맞다면 2뎁스인 option필터링을 한번 거친 값을 들고 있어도 좋을 듯합니다.

1.
```
const filteredItemsByOption = rankingItems.filter(({option}) => option === selectedOption);
const renderedItems = showAllRankingItems ? filteredItemsByOption : filteredItemsByOption.filter(({category}) => category === selectedPersonCategory).slice(0, 6)
```

2.
```
const filteredRankingItems = rankingItems.filter(item => {
  const isCategoryMatch = selectedPersonCategory === 'All' || item.category === selectedPersonCategory;
  const isOptionMatch = item.option === selectedOption;

  return isCategoryMatch && isOptionMatch;
});

const renderedItems = showAllRankingItems ? filteredRankingItems : filteredRankingItems.slice(0, 6);
```

Reply:
추가로, 변수는 setState로 관리하고 상태 간의 변화를 `useEffect`를 통해 업데이트하는 것이 좀더 리액트 생명주기에 맞을 것입니다. 

useEffect없이 상태를 관리하다보면 무한루프에 빠질 위험이 커지고, UI상 렌더링되지 않을 수도 있고, 컴포넌트가 렌더링될 때마다 해당 변수를 새로 만들기 때문에 불필요하게 연산이 커집니다. 아래 코드를 참고해서 useEffect 의존성배열에 필요한 상태들을 triggering시켜서 상태를 업데이트해보세요. (일단 호정님이 작성하신 변수를 그대로 useEffect방식으로만 바꿔보겠습니다)

useEffect는 여러개 생성해도 됩니다~!

```suggestion
  useEffect(() => {
  const newFilteredRankingItems = rankingItems
    .filter((item) => item.category === selectedPersonCategory || selectedPersonCategory === 'All')
    .filter((item) => item.option === selectedOption);
  
    setFilteredRankingItems(newFilteredRankingItems);
  }, [selectedPersonCategory, selectedOption, rankingItems]);
  
    useEffect(() => {
    const newItemsToShow = showAllRankingItems ? filteredRankingItems : filteredRankingItems.slice(0, 6)
    setItemsToShow(newItemsToShow);
  }, [showAllRankingItems, filteredRankingItems]);
 
```